### ISSUE TITLE
[Offer] Commercial Release

### ISSUE THREAD

**Issue by Tormentor667** [Realm667/WolfenDoom#282]
About 2 weeks ago, I got an offer from a game development company. Currently I can't tell you the name of the company neither details about a possible contract for secrecy clauses but it's something we need to discuss, as everyone who is assigned to this issue has shared larger or smaller portions to the mod. Considering that each of us has invested more or less time to "Blade of Agony", it's also a copryright question, that we can look at closer as well.

**So what is the offer all about?**
The game development company (short gdc) has offered to turn our standalone mod project into a full game that will be released on pc but also on other platforms (Nintendo, XBOX, Playstation, and many more that are able to run the engine). As we are using a lot of copyrighted material from all sorts of games, the company offers to recreate all (!) copyrighted assets with original material. Yes, you read right. All of the material from other games will be replaced - sprites, textures, sounds, music, etc. The company will also take care of the marketing, the advertising and everything that needs to be done to make a commercial release possible.

**And what do we get? What do we have to do?**
50/50. Every single dollar gets split up between the gdc and us, the primary development team. Our role during the replacement and finalizing of the commercial release will be in a consulting way, and we have the final word when it comes to important decisions. The gdc will take care of everything else. The contract also says that we will still be able to release "Blade of Agony" in its final version as a mod - as it was intended. The gdc on the contrary will give it a new name (where we also have final word about) and take care of the commercial release on all platforms then, including payments.

**How is the profit shared?**
Profit will be shared regarding the amount of investment to the final product in a percentual calculation. The core team (@DoomJedi , @doomjuan , @AFADoomer , @Ozymandias81 , @Tormentor667 ) will each 15% of the profit, the primary support team will get 10% (@Talon1024 ), the secondary supporting team (@nashmuhandes , @madame-rachelle , @Username-N00b-is-not-available , @Guardsoul ) will each get 3%. The remaining 3% will be used for everything that's necessary when it comes to advocate fees (a security setup) just in case we need that.

**When do you have to decide?**
At the moment, we have no real time pressure. "Blade of Agony" is and stays a free product, we will stick to the philosophy, and that release has the most priority at the moment. But the sooner we have discussed a decision, the sooner the company can start working. So let me know what you think about it, I am looking forward to your thoughts. Nothing is decided, nothing is set in stone.

Best regards,
Dan

-----------------------------

Updates regarding the NDA between the members and me. I have personal information currently from the following people and I need all of them to sign the NDA between me and the company.

- [x] @DoomJedi 
- [x] @doomjuan 
- [x] @Guardsoul 
- [x] @AFADoomer 
- [x] @Username-N00b-is-not-available 
- [x] @nashmuhandes 
- [x] @madame-rachelle 
- [x] @Talon1024 
- [x] @Ozymandias81 
- [x] @wildweasel

**Comment by Doom2fan** [comment]
This sounds like a scam. Why would they recreate the assets for free?
Additionally, GZDoom cannot be used for console games on most consoles. The Doom source code is GPL-licensed, and so is GZDoom. The GPL requires any changes to the source code in distributed binaries to be made public and licensed under the GPL, which directly conflicts with console development, as all three major consoles (Switch, Xbox One and PS4) all have very strict NDAs on their development SDKs and APIs - which means releasing a GZDoom game on one of those consoles requires breaking laws, be it by breaching the NDAs or by breaking licenses.

**Comment by Tormentor667** [comment]
Thanks for the information, this is certainly interesting, and I am going to confront the developers with this. Still, I am eager to hear what ohers think.

**Comment by madame-rachelle** [comment]
It's feasible. With a cost/benefit analysis, someone ran the numbers and thought it would be marketable. I'm not saying phantom is wrong - you should definitely be skeptical of things like this - but the flipside is - corporations have literally unlimited resources to solve these issues - so a realistic possibility of this is there. I think the most relevant thing at this point is - if the game company's name can't be revealed, then the question becomes how well known are they? Do they have a reputation? If not, then yeah, it is likely a scam.

So with that under consideration - this becomes a cost-benefit analysis for this team. What is there to gain vs what is there to lose. This project obviously has years of hard work invested into it so there's no way something like this could happen unless all the people who've spent their precious time into this are okay with it. But at the same time - there is that. Obviously if this offer is real, what will get preserved are the gameplay decisions and the coding, which likely will see little change in the final product. And whoever created all that are the ones who has the most to lose if this is a scam.

Do remember though - life is all about risk. If you risk nothing you gain nothing, and that is especially true in this case. You could say no, and be safe - but what might have you missed out on?

Again - I think the most important thing at this point is - how reputable the source is - and how much you can trust your contact to be who they say they are.

**Comment by Tormentor667** [comment]
For a better evaluation, the game studio has released over 10 games that look quite cool and they also helped with collaboration work on God of War, League of Legends, The Last of us, Uncharted but also legwork for Sony and Panasonic (according to their website). I can't tell with what they helped in detail though. The team consists of 16 programmers and game developers and was founded in 2006.

The point why I currently can't reveal it, is, that I don't have clearance for that yet. I will ask the founder if I can discuss this in public and then I can let you know what company it is and also post the link to the webpage.

**Comment by madame-rachelle** [comment]
Then the only issue left is whether you can verify that the point of contact is who they say they are - and if they are, then it sounds like a very exciting opportunity.

**Comment by madame-rachelle** [comment]
As for GZDoom being GPL - unfortunately it does seem like a lot of the internal "guts" of the game will have to be retooled - but if they have the programmers for that, then that might not be an issue on this team's end to deal with. That's up to their coders and legal department.

**Comment by Username-N00b-is-not-available** [comment]
I'm not nearly such experienced as you, ladies and gentlemen, and know almost nothing about Western laws, so you may disregard completely the following dumb things that popped into my head after reading your posts. I was asked to respond however, so I shall start.

1) Do you remember the story of TNT: Evilution? TeamTNT started it as a regular megawad, but it was 'bought', and as doomwiki says, this caused 'a certain amount of controversy' in Doom community. See also https://www.doomworld.com/10years/bestwads/infamous.php and https://groups.google.com/forum/?hl=en#!msg/alt.games.doom/Fhfl5PHTbVk/0eSgX8fkDJcJ (link from that page).

2) The GDC says that they would like to replace every copyrighted resource. Well, okay; have they really seen the amounts of data we have in the archive, those 387 folders and 26K files in them? Would their composers have enough time to write **one hundred music tracks from scratch** (well, maybe a bit less because of jukebox songs removal)? What about **562 3dmodels**? **>3K textures?** (The sprites have been redone in bulk by @DoomJedi, so this shouldn't be a serious problem.) Does the company really have so much manpower to replace **all** the resources? If by chance it turns out that they don't — for example, they get a more high-budget game which needs a lot of work, — then the development will stall. And in such situation, with all the rights delegated to the GDC, our team could only cry and wait for the release, as if the core team members haven't devoted _thirty years of their lives_ in total to the project.

3) Finally, Blade of Agony has belonged to the 'extended Wolfenstein universe' since its very start in 2015. So, unless GDC == MachineGames (unlikely because it was founded in 2009), what will they do with the game universe? With direct references to Wolfenstein? Who could replace B. J. Blazkowicz? What changes will be made to the story, possibly even to the political worldview carried by our game (very subtle, but we do have one)? I think nobody in the team wants this project to become another 'Generic FPS'.

Again, I'm not a figure in neither Doom nor general gamedev community, I present my point of view here only because I was asked to. The core team of the project is of course free to do what they want with their work.
To all the people at the GDC: please don't take this personally, at the time of writing this post I don't even know who you are. I may have exaggerated the possible problems, but I do not oppose this idea at all (and 3% from a multiplatform game which could sell at least tens of thousands of copies is quite a temptive offer :-) ). I just wanted to warn the 'core team' once more that they are **obliged** (no pun intended) to think about this **very** carefully <strike>so as not to turn 3% to 30 pieces of silver</strike>.

**Comment by madame-rachelle** [comment]
Okay - on point 1) the controversy... no matter what anyone does these days, there's controversy. Have you seen how divisive it is to simply stand for humanity against fascism? It's not my intent to make this political - but the main point I am trying to make here is there will always be haters, and the fire will always be blown out of proportion no matter what. What is important is sticking to morals and what's right - but bending to the will of people expecting you to make content for them for free and promising not to make money off of it is utterly ridiculous and insane. The whole controversy was a shitshow then and it's a shitshow now.

2) Is a very valid concern, though, this is a ton of work for 16 people. These people would have to be very talented to pull that off - and even then, it also calls into question the ethicallity of making them work that hard for it. I know it can be done - I just hope it can be done sanely and with a "work smarter not harder" mentality without cutting corners too much in quality.

3) Also a very valid concern - technically you could just change the names of all the characters and call it a day, but it may lose a lot of its "magic" in the process. Not as concerning as point 2, though.

**Comment by Talon1024** [comment]
> This sounds like a scam. Why would they recreate the assets for free?
> 
> Additionally, GZDoom cannot be used for console games on most consoles. The Doom source code is GPL-licensed, and so is GZDoom. The GPL requires any changes to the source code in distributed binaries to be made public and licensed under the GPL, which directly conflicts with console development, as all three major consoles (Switch, Xbox One and PS4) all have very strict NDAs on their development SDKs and APIs.
> This means releasing a GZDoom game on one of those consoles requires breaking laws, be it by breaching the NDAs or by breaking licenses.

If GDC wants to release on consoles, they'll either have to get a special license from Bethesda/ZeniMax Media, in addition to permission from **all** of the GZDoom contributors, or they'll have to **rewrite the entire game engine from scratch**.

If they were to attempt the former option, tracking down each one and asking for their permission is highly unfeasible, since GZDoom has a VERY LONG history that predates the existence of version control systems like git, and thus it is difficult to know all of who wrote which pieces of GZDoom code, particularly when you're talking about the core Id Tech 1 code, which evolved over the course of 20+ years of bug fixes and feature additions.

If they were to attempt the latter option, I'd wish them the best of luck with that, since that may take an insurmountable amount of time. Id Software's modern console ports of Doom use Unity3D as a minimal wrapper around the console platform APIs. The closest example to an engine rewrite I can think of is Nightdive's reworked engine for [Strife: Veteran Edition](https://store.steampowered.com/app/317040/The_Original_Strife_Veteran_Edition/). But, it's open source, and I have no idea whether it's available for consoles or not. Additionally, Strife: Veteran Edition's engine is based on the Chocolate Strife code and Nightdive's "Kex engine".

In addition, to think any company would offer to replace ALL of the assets in WolfenDoom is.. suspicious, to say the least. I've never heard of any such thing being done throughout the entire history of video games. It also sounds as if GDC is expecting us to throw in the towel, hand over the keys, and leave the rest to them. I take issue with that because this is the culmination of *5+ years* of work in our spare time, even if it isn't 100% our own work. Is it right to just throw it all away for money? What if GDC adds some scummy predatory monetization schemes in the process, like many AAA game studios are known to do?

**Comment by DoomJedi** [comment]
I have been following this discussion, and I guess it's my time to step in.

It's a pity big chunk of this discussion was wasted about...some fictional proposal probably, and not the actual proposal described in the original post.
What CONTROVERSY you're talking about? What RISK?
The commercial offer is IN ADDITION to normal BoA mod release, nothing changes to the BoA mod itself - release, it being free, or anything else.
What risk is there to approve RE-USE of the mod as commercial version - for work already made? It's free money for no extra work dudes (and dudettes :) )
So as long as legal issues solved, and the company will be fully aware about what it's getting into (copyright-wise, scale of rework needed etc) - why would anyone oppose or see it as a risk? If fails - then only the company will have losses, risk is only on their side. So this is a very kind offer, if it's not a scam. No pain, only gain.
Because of such - investment and risk being only on one side - I think if the scope of rework will be much larger for them than originally thought (by them, including need to rewrite the engine) - we can negotiate a lesser deal with the company - with lesser percentages. As it's still free effortless ,risk-less money for all (of us) involved, and the more work they'll need to do - the bigger profit cut they should earn, it's only fair.

And now, after I said this - let's talk about the serious problems and challenges that I do admit to exist.

As far as I'm aware - almost no resource of this mod was made from scratch, from zero. It's either direct use of copyrighted resources, or edits of those resources, some more extended, some barely. To add to that - whole Wolfenstein plot/theme is copyrighted. Not to mention the secret maps - Keen, Astrostein etc.
Some wrote that at least my sprites can be used as is - I'm not a legal expert of no sort, but it seems to me it's not the case.
I didn't make any or almost any art from zero. Not to mention many art is uses "as is" or with very little edits. From Mechas (from Greed) to Monks (from Blood, with added extra details).
Enemies - they are still based on commercial base (poses, walking, death etc), scaled and edited - legs in particular are just primitive scaling filter of commercial art (as we rightly assumed noone looks too much on enemy legs in fight).
We use MAC Bosses art (with or without small edits), MACres  remakes of Bosses from Wolf3D mods (like Trench Warfare) etc.
I can go on but it's enough to make a point.
To add to this - what about art quality and consistency? Can they make professional art replacements to some - and leave the others as is? As is - I don't think my art stand by commercial quality standards, they are fine for a mod, but....

To balance the scale of rework needed - I want to note that they don't need to make all the new textures, sounds and music from zero. There are many free texture packs, sound packs and music available - even for commercial use (just as Doom makers used free sound libraries for Doom). Most sounds and art are quite standard to WWII thematic.
And others - can be bought by much lesser money (and time) that it would take to pay to a person to develop. So it's not THAT of a big deal guys, our resources are not THAT unique to the most part, especially in such historic and realistic setting.

Now if some sprite art is to be kept as is - credits and permissions. How will we track to endless makers of the various resources?
Will we ask Ringman for his permission for art made by him? Ask Astrostein maker for his? What about rights for Tomb Raider or Indiana Jones? etc etc? This looks endless...how can one keep track of all the original resources reused in the mod?
This mod is a mess commercial-wise...was really made without any direction of future commercification in mind....

**Comment by madame-rachelle** [comment]
> What CONTROVERSY you're talking about? What RISK?

I was responding to this post: https://github.com/Realm667/WolfenDoom/issues/282#issuecomment-684034309 - which, I'll be honest, I think it does greatly exaggerate the issues it states, rather than making much of a mention of the real issues, such as what you and several others have already touched on.

**Comment by DoomJedi** [comment]
I agree, but he stated humbly upfront:
"I'm not nearly such experienced as you, ladies and gentlemen, and know almost nothing about Western laws, so you may disregard completely the following dumb things that popped into my head after reading your posts."

**Comment by madame-rachelle** [comment]
Sorry - but I did feel like the issue had to be addressed. I feel that any risk of "controversy" for accepting money should be completely and utterly disregarded. There are some very hyper-moralistic views in the greater Doom community that I think greatly damage the community as a whole - and the whole notion that no one should get paid for a hobby is one of them.

**Comment by DoomJedi** [comment]
I fully agree.

**Comment by Ozymandias81** [comment]
Here I am with some words to spend with this indeed good news, but which requires deep thoughts and focus on what could be involved in all of this. While I think I understood which company it is, and they indeed do some good games, there are some limits described on their website in legal terms but even open for people who wants to create mods with their games, all for free with a granted IP for modders but that can be shutdown anywhere anytime (this is something new and makes me think why this offer sound so good, since they seem very open to these kind of things in the recent years). They also clearly state no ripoffs, only original content to be included on mods done with their games, and for what I see they never released an fps under GZDoom or any GPL engines.
The pros for me about this offer is that it seems they will be fully eager to have our mod released as it is, and that we'll gain 50/50 in terms of profits once we will be all okay with this offer and when the whole thing will start (I guess profits will happen only once the game will be fully developed and released, or at least once an Early Access version will see the light, on Steam much probably since they are much present there - also GoG). I also think the question about releasing the game under other consoles could be something optional, since they didn't make many games for consoles (if I guessed right which company it is).
The cons instead are a bit complicate to list but let me explain: what will be exactly our role during game development? what binds us to have such 50/50 in lawyer terms, how's that possible? what would will cost us in terms of advocate fees, if something else could happen? I worked on most of the models present on BoA, and many of them comes from CoD games, RTCW, Wolfenstein 2009, Skyrim, Necrovision, MoHAA just to say and what could happen legally for the gdc but also how much could be involved us too? We surely don't have a team of lawyers like them, maybe they could be safe and surely know howbtgese things works, but what would be of us and the mod itself? Remember that it is hosted on a website belonging to DGMedia and Realm667, so surely Torm would have some to take care carefully. Then there is this fact about that they will replace all resources with brand new art/material... Will they use still GZDoom? Do they have plans to create 2d or 3d resources? How much their game will be similar to our mod? I also guess that all these things will happen only when we'll give all fully agreement and when the mod first will be released.
There are so many questions that it may take months from me to ask and theorize, and while everything sounds very nice and respectful versus our work and time spent for BoA, indeed we first need more info about what are terms, role, ideas, work and plans from the GDC versus the mod, their game and the mutual collaboration.
I am eager and open for a commercial release, which I am sure will look very different for many reasons, and in case I may help to provide a sort of list of copyrighted material present on BoA versus free license and brand new.

**Comment by DoomJedi** [comment]
I have to note - 50/50 on profit - I assume only starts when/after they returned themselves in sales what they have invested into it... How will we know how much they actually invested in it overall? Development, marketing...all is very expensive nowdays. And how will we know the actual profits across all platforms?

"and for what I see they never released an fps under GZDoom or any GPL engines."
They could have some fitting engine on their hands though, already ready for repurposing...but will maps be compatible - or remade?

**Comment by Tormentor667** [comment]
First of all, thanks to everyone who has joined the discussion for sharing your points of view, your thoughts and your opinions. A lot of good approaches have been made, a lot of good reasons to turn this into a real thing and a lot of good reasons to avoid even considering this.

The **license** issues is something I am not aware of. I have no idea what needs to be done to make the engine work on a commercial base on other platforms, I only know that 99% of the mod's content has been drawn from other sources and needs to be fully redone - 27.000 files to work through. Ambitious, doable but ambitious. I wrote an email regarding the licensing to the main developer and I am still waiting for an answer.

Now let's talk about the **profit** in general. To be honest - and this is where I am sharing the attitude of other people and where I already said that I don't want to end up like TeamTNT - the whole mod is a hobby. I don't want to earn mony with it, I don't care, it's just for the fun of it. It keeps me creative, I like how everything turned out lately and I am proud of what we have achieved here - everyone of us, this includes the whole Doom community. Seeing that professional gaming companies get interested in this proves that we are doing a good job. But it's not for the profit. I have a good paid job for my living, no need to force hobbies to create some earnings for me.

But why do I still **consider a commercial release**? Remember: Blade of Agony gets released as a standalone mod without any fee, this will happen no matter if a commercial release gets developed or not - promised! What I kinda find interesting is to make the game available on other platforms. And to have a professional gaming company creating assets and resources for the complete game. I wonder: How would it look like? What is the company capable of making out of this standalone mod? That's definitely something I am curious about. And that's why I was considering this.

Anyway, it stays and falls with the solutions the team has regarding the license questions. If we get a proper answer, I will share it with you and we can continue deciding and discussing.

**Comment by DoomJedi** [comment]
I assume you know we are not there for the money - it was free mod all along - at least till this offer came just now. We all have been committed to this, without any profit promise, never intended or hoped - even for pure copyright/legal issue. 

But now - I don't see why we should give up this opportunity - even for profit reasons alone. I have full-time job too, and am not in need of money (though there is never too much money in life - my approach, I'm far from being rich), but I do welcome extra income.

This is the difference between us.
I don't need more seasons to agree (as long as free mod is still released, with my art) - than the profit hope alone.
I respect your approach, just trying to be sincere. I am dreaming of indie games (in terms of putting anything for sale) for quite some time now (even tried to make such with TR coder on a few occasions, didn't go too far unfortunately), and I know I'm not the only team member that has those dreams - to make some profit out of my beloved creative hobby. Not to mention my wife's pressure to make some profit of my hobby, heh :)

I respect you approach, just trying to be honest about it, from my perspective.

**Comment by Tormentor667** [comment]
If we get the opportunity, to legally and in total agreement of all contributors to earn some money with it, I definitely won't say no :)

**Comment by madame-rachelle** [comment]
I fully agree with DoomJedi's views on this. From what I see there's little to lose and a lot to gain.

**Comment by Tormentor667** [comment]
Just for the record: No reply yet.

**Comment by Tormentor667** [comment]
I just got a reply to all my questions when it comes to the commercial offer, so here they are for you to read as well. My questions are _italic_ and the answers from the company are **bold**.
_____________________________________

_1. One question I have is when it comes to discussing everything with the team behind "Blade of Agony": Can I publicly tell that your game studio is interested in commercializing the game?_
**Of course you can tell anyone about our mutual collaboration as soon as we both sign the NDA. We will also do the same.**

_2. I was talking to one of my teammates, who is more used to the technical and license background informations, and this is what he said:This sounds like a scam. Why would they recreate the assets for free?_
**We are not doing anything for free. We expect to get paid as soon as the game gets released in its final form. The assets will not be for the free mod version, so we are not giving away any of our work for free.**

_3. Additionally, GZDoom cannot be used for console games on most consoles. The Doom source code is GPL-licensed, and so is GZDoom. The GPL requires any changes to the source code in distributed binaries to be made public and licensed under the GPL, which directly conflicts with console development, as all three major consoles (Switch, Xbox One and PS4) all have very strict NDAs on their development SDKs and APIs. This means releasing a GZDoom game on one of those consoles requires breaking laws, be it by breaching the NDAs or by breaking licenses. So how do you approach this problem?_
**Bethesda already circumvented this limitation by using Unity engine as a mediator. There already exist many level loaders for wad files in unity. After all, as I already told you from the very beginning of our communication, the ports to consoles will only happen should we get enough money from the sales of the PC version. GZdoom also lately changed their way of handling things and standalone games are already appearing because of these changes, such as Hedon which is already being sold on steam.**
_____________________________________

So let me know what you think about the answers and if there is anything else we might ask the game studions.

**Comment by Doom2fan** [comment]
> **Bethesda already circumvented this limitation by using Unity engine as a mediator. There already exist many level loaders for wad files in unity. After all, as I already told you from the very beginning of our communication, the ports to consoles will only happen should we get enough money from the sales of the PC version. GZdoom also lately changed their way of handling things and standalone games are already appearing because of these changes, such as Hedon which is already being sold on steam.**

This is incorrect. ZeniMax didn't circumvent anything by using Unity as a mediator, they did that for convenience. They fully own the rights to the Doom source code and do not need a license to be allowed to use it, which means they don't need to obey the GPL license used in the linuxdoom source code release.
Using Unity to circumvent the GPL's requirements like this is impossible, as it specifically forbids linking libraries that aren't GPL licensed and being linked by programs that aren't GPL licensed (aside from very specific cases, like OS components). Doing so constitutes a breach of the license.
The right to GZDoom's source code, meanwhile, is owned by several different people and also by ZeniMax itself. You'd have to get permission from ZeniMax and every single person who has worked on or had their code used in GZDoom to be allowed to do this, which would be extremely hard to do.

This part of the response makes it even seem more sketch to me, as it sounds like they only had a quick glance at info you can find on Google, rather than actually checking any of this with someone who knows anything about licenses or a lawyer.

**Comment by madame-rachelle** [comment]
While phantom is right - let me be specific about one thing.

If they release the source code to their GZDoom - they're in the free and clear, as far as GZDoom licenses go.

That leaves things on the console end. GZDoom may require *very specific modifications* in order to work with consoles - *however* if the modifications are only to allow it to work with *already present __system__ libraries* (and they do not have to create intermediaries to use them, or they are allowed to keep them open source) then this is a much more feasible bridge to cross than we might think.

Basically, if they can get GZDoom to work on a console but keep the source open while doing so without violating the console's NDA - then they are fine. This might limit them to XBox and PlayStation, since they use already-established operating systems for their back-ends, but a Nintendo is probably not strong enough to play a mod like this, anyhow.

Using Unity as an intermediary would not work in this case - I don't think Unity even has the ability to work with GZDoom's code like it did with the DOS Doom's code, and even if it did, the entire system had to be rewritten for Unity itself in Bethesda's case - none of the backend code which made the DOS game function is present at all in the Unity version.

But again - let me re-iterate - if they are willing to take this on - I think it is in our best interest to just let them try.

**Comment by DoomJedi** [comment]
...so how Hedon handled those issues?

As I see it - we have 2 options:

1. To tell the company the possible legal "mines" they might have been missing, in weird "hope" to convince them to give this up.

2. To tell them - "after we raised our concerns and warnings about GZDoom copyright/legal state - we are ready to move on with this, but just to be stated clear - from now on we are not responsible for any legal issues that might arise with making of the commercial version, and every such issue will be up to the company to deal with. Company answers so far didn't address all of our related concerns regarding the issue (we are ready to tell those by request though), but as long as it's not our problem - we won't mind" 

Why the hell should we worry more about the legal side - than the company itself who gonna invest hard into this??? The company has probably a legal department to deals with the legal side of this.

**Comment by Talon1024** [comment]
Hedon didn't make extensive changes to the GZDoom code, and the modified source code [is available here](https://github.com/madame-rachelle/hgzdoom). Even though Hedon is available on [Steam](https://store.steampowered.com/app/1072150/Hedon/), [GOG](https://www.gog.com/game/hedon), and [itch.io](https://zanthebonebreaker.itch.io/hedon), Hedon doesn't use any of the proprietary APIs (and thus features like achievements) associated with storefronts like Steam or GOG.

**Comment by Ozymandias81** [comment]
So the same happened also for The ForesTale? I have inspect a bit Hedon files and noticed some leftovers here and there of codes and credits which shouldn't be present, so I guess in these cases we mustn't be extremely nitpicky... Well but I am nitpicky though ahaha
There are indeed several concernings regarding the legal side, but we should trust then this company which plans to make a game out of our mod, in any case they first plan to release on PC and see how the game goes after its final form. A form that I still want to see how it will be though, as Doomjedi and DoomJuan said on our private chats.
The question regarding console port imho is a secondary scenario that, in any case, will be up to GDC to take care since for that time we'll be done with our contribution and game should be 100% ready for PC on digital stores. What I really want to know is: which engine they plan to use, Unity? GZDoom? Something else? Also what are their plans for the game, how much different will it look? What they will do with eastereggs too, which are one of the most important side of our mod in terms of "spirit"?

**Comment by DoomJedi** [comment]
What would be "Brutal Fate" status btw....STG Mark plans to put it on steam fro sale, and it's gzdoom mod...

**Comment by DoomJedi** [comment]
A few more possible questions:

1. How would they handle the Wolfenstein/BJ related plot? Did they play enough of BoA to see how deeply it's Wolf3D-lore based? Plot-wise etc. Not to mention copyright cameos and easters. 

2. How can they start developing the new game based on BoA right away - while BoA is not finished and keeps adding stuff (like merchant just recently).

**Comment by nashmuhandes** [comment]
My only question is why don't they just put together a new, original GZDoom-based game from scratch, it would be much more practical to make a fresh new game than to reskin BoA, if you ask me... :D

**Comment by DoomJedi** [comment]
....yeah, let's ask them that, money is overrated in this world anyway..... :)

...It's part of more general question - why us? Out of all the mods? From "Total Chaos" to "Brutal Rage"...

**Comment by Tormentor667** [comment]
I just got an email response from the studio leader, and well, it is still a very convincing offer for me, it simply is something we could gain from, and we have nothing to loose, as @madame-rachelle has already stated numerous times. Let me share that with you:

> The whole proposition was on the basis that we as studio are avid gamers who have been following Blade of Agony for some time, while we are also witnessing the emerging trend with retro FPS games. In no way, shape or form do we want to be the bad guys here, that are placing their greedy hands on a free Doom 2 MOD and trying to make money out of it. If there are people on your team disagreeing with commercializing the project, this is something you will have to discuss with them.

That most propably answers your question as well @DoomJedi 

> However, _if most team members are onboard and willing to commercialize Blade of Agony_, these are the possibilities:  If a team member’s work is not so crucial for the project, meaning that it won’t take us months to remove/replace their work, e.g. the disagreeing member is not your lead programmer because that would mean remaking every game interaction or most of them, then we can proceed as originally planned and just remake their assets (ideally, you should sign NDAs with all your team members so that they are covered that from your 50 percent of the sales of the game they will get their cut and that they can't do anything in any court of law against you should they change their mind at a later date. You should also sign NDAs with any disagreeing member so that if the game gets released commercially and all their work is replaced they won't come asking for money or whatever.)
_If most members of your team flat-out disagree_ and you don't want to have to fight or explain to them the possibility of making money out of it, you may proceed as originally planned with the free mod release of Blade of Agony and we can cooperate on a different basis. After all, the whole proposition was based on your excellent mapping work and not on the work of the rest of your team, which we don't know.

So Blade of Agony could happen as a commercial release, but this can only happen if the core team members, who invested most of the content, code, mapping, concepts and resources agree on commercializing it. For other contributors, it would help to sign an NDA so we can use their work, but it isn't necessary if the studio can recreate the assets and work.

**So how is this going to happen?**
The studio told me that they want to sign the NDA with me as project leader. They told me, that I have to make NDAs with the team members then to make sure that everyone who contributed to the project gets what he deserves when it comes to profit and praise, but also to make sure, that I won't end up in jail literally.

The studio's teammates are paid by the team, which is not drawn from the profit. 50% of the profit will be given to me where I am sharing the profit as proposed with everyone involved as stated in the initial post of this issue. Taxes  and intermediate expenses are paid in advance according to the studio.

Regarding the licensing, this is what they said:

> Since we have an NDA with you and you agree on the commercialization, the GPL etc. is covered by us and you shouldn't worry about it. [game studio] is in front of everything so that no sole person takes responsibility and we are all covered behind this legal entity.

So, no matter how the licencing is going, it's the studios responsibility, not ours.

**Lots of talk, what do we do now?**
We have nothing to loose. The mod release will still happen as we have planned it, the studio takes care of everything and the only thing we need to take care of is preparing and signing the NDAs and agree (or disagree) on commercializing the mod. But I can't make the decision on my own, so I guess, people who are questioned now are you:

The core team (@DoomJedi , @doomjuan , @AFADoomer , @Ozymandias81 , @Tormentor667 ) will each 15% of the profit, the primary support team will get 10% (@Talon1024 ), the secondary supporting team (@nashmuhandes , @wildweasel (as proposed by @madame-rachelle ) , @Username-N00b-is-not-available , @Guardsoul ) will each get 3%.

So now it's time to decide if we are brave to go for that.

**Comment by madame-rachelle** [comment]
I am on board. I may not be in the core team but I still say let's do it. This is an opportunity you don't get often, and imo, it would be a little insane to pass it up. That being said, I do have things I have to complete on this project as well and I am still planning on doing it, but now that I see the studio's response I realize it's much more important now.

**Comment by madame-rachelle** [comment]
FWIW I am not seeing a lot of red flags with this. Even if there were there's little to lose for a lot to gain. (sorry for 2nd in a row post)

**Comment by AFADoomer** [comment]
I'm good with it.  

Definitely agree with at least offering wildweasel a cut - despite the fact that we've replaced huge swathes of the original mod, we wouldn't have anything without his original work.

Caveat:  We need to ensure that BoA still gets released in mod form above all else.

**Comment by DoomJedi** [comment]
Last question is very important to me as well (but I care only about my sprites, not the rest), as I really want my current BoA art to be used in Wolf3D/Doom modding - after BoA release. Many are anticipating that.
Clarification on that would be nice, but I assume it's fine - as my art will probably be replaced for commercial version anyway (would be VERY cool if not though, but not much hopeful due to possible copyrights and art consistency issues) and will only be part of free BoA release. And their new commercial art will obviously be modding-protected anyway (though still probably will get modded, heh, but not my business).

But yes, I'm fully onboard, this is exciting development I did not ever dream of (but would - if I'd think back then it's possible).

P. S. - Why we are talking (only?) about NDA regarding profit? Isn't it just a secrecy agreement for developers and such? Or is payments/share are part if it as well?

P. S. 2...so it's only about Daniel's impressive mapping work they care about, hah....my ego as a spriter is a bit hurt (to be honest)... But I'm happy for Daniel, it's great achievement for him. Congrats pal :)

**Comment by DoomJedi** [comment]
I few possible questions though - not yet addressed:

1. How they plan to deal with Wolfenstein IP
2. How will we know their exact profit to make sure we get 50%?
3. 50% of profit - is profit defined "after development costs are covered" - or all income from it?
4. Just to make sure - our agreement covers all possible console versions in the future? Not only PC one?  I assume it does, just to make sure.

**Comment by nashmuhandes** [comment]
@Tormentor667 here's my final answer: Ok, I accept.

**Comment by Tormentor667** [comment]
Got word from wildweasel: 

> I had skimmed over it after a contact of mine let me know about it, yeah. Honestly I have my doubts so far, but it doesn't seem like my business to question it, especially not when NDA prevents me from knowing the details.

**Comment by Ozymandias81** [comment]
I also accept too, and @doomjuan sent us on facebook team chat the ok for it. We need now @Guardsoul reply yet.

**Comment by Guardsoul** [comment]
You have my approval. No complaints on my side. Go for it!

**Comment by Tormentor667** [comment]
Fine. Next step is to provide the NDAs between me and you and to adjust the NDA between me and the company.

**Comment by Tormentor667** [comment]
Okay, I got news and i think we are ready to go. With the help of the company, I prepared a NDA between all the teammates and me that will be the basis for the NDA between me and the company. 

I would like to send the NDA to everyone involved, so you can read and sign it, and then send it back to me. For this, I need some information from you though so please send me an e-mail to info@realm667.com with the following data

- Full Name
- Street and Number
- Postal Code and City
- Province
- Country
- E-Mail Adress

from the following contributors: @DoomJedi , @doomjuan , @AFADoomer , @Ozymandias81 , @Talon1024 , @nashmuhandes , @wildweasel , @Username-N00b-is-not-available , @Guardsoul

**Comment by nashmuhandes** [comment]
@Tormentor667 done

**Comment by Ozymandias81** [comment]
@Tormentor667 I did the same as well, protected e-mail just in case so 1 month to read it.

**Comment by DoomJedi** [comment]
email sent

**Comment by Guardsoul** [comment]
The email has been sent as well.

**Comment by madame-rachelle** [comment]
Sent mine.

**Comment by Talon1024** [comment]
Just sent my e-mail

**Comment by Tormentor667** [comment]
@doomjuan I am still missing your NDA information.

**Comment by DoomJedi** [comment]
Wrote to him on moddb....nothing. Frustrating.

**Comment by Tormentor667** [comment]
Got his mail, time to prepare the files.

**Comment by Tormentor667** [comment]
I did answer actually, maybe my answer got spam-filtered?

**Comment by Username-N00b-is-not-available** [comment]
Very strange, my spam folder is empty and AFAIK has been for the entire week. Do you mean you replied to my letter which starts with "Hello again, Tormentor667! It is pleasing to hear that...", or to another one?

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 Very strange, my spam folder is empty and AFAIK has been empty for the entire week. Do you mean you replied to my letter which starts with "Hello again, Tormentor667! It is pleasing to hear that...", or to another one?

**Comment by Tormentor667** [comment]
I will check this tomorrow, but can you be so kind and send yours again?

**Comment by Username-N00b-is-not-available** [comment]
Sent it again.

**Comment by Tormentor667** [comment]
Got it, will reply tomorrow.

**Comment by nashmuhandes** [comment]
Sorry for the bump, but just checking what's the status with this? I have not received any e-mails since the last conversation here.

**Comment by Tormentor667** [comment]
Currently I can not sign a contract until @AFADoomer has been able to agree with the NDA I sent him.

We have already been in contact via email, I think as soon as he has enough time to read through that, he will send me his answer.

**Comment by Tormentor667** [comment]
I just sent the NDA to each of you, waiting for your reply now :)

**Comment by Tormentor667** [comment]
4 NDAs are missing, the rest arrived via E-Mail.

**Comment by AFADoomer** [comment]
I have to re-send my comments because your mail server thinks I'm sending spam again.

**Comment by Ozymandias81** [comment]
Sent my signed NDA a while ago.

**Comment by Tormentor667** [comment]
> I have to re-send my comments because your mail server thinks I'm sending spam again.

That’s strange, I put your whole domain and email on my whitelist after the last time:(

**Comment by Tormentor667** [comment]
**Update**
Send you another mail @AFADoomer with an updated NDA.

**Comment by nashmuhandes** [comment]
@Tormentor667 woops! I missed the e-mail. Will e-mail you back ASAP!

**Comment by Tormentor667** [comment]
> @Tormentor667 woops! I missed the e-mail. Will e-mail you back ASAP!

Better late than never ;)

**Comment by Tormentor667** [comment]
@nashmuhandes @AFADoomer just for the record, I didn't get a email yet, wrote one to you @AFADoomer yesterday regarding your thoughts on the NDA.

**Comment by nashmuhandes** [comment]
@Tormentor667 have already e-mailed you. :)

**Comment by AFADoomer** [comment]
@Tormentor667 - I just sent the e-mailed the signed document back to you.

**Comment by Tormentor667** [comment]
Will check it this evening

**Comment by Tormentor667** [comment]
Thanks to @AFADoomer and @nashmuhandes , got it. Only @Username-N00b-is-not-available is missing now.

**Comment by Username-N00b-is-not-available** [comment]
I'm sorry for the delay in my signing the document and I hope that my circumstances will make it possible in the next few* days.
*should not take more than a week...

**Comment by getphobo** [comment]
Hello,
I'm curious what this all would mean for the people providing translations to BoA.
I'm happy to provide free work for a free community project, but if it's supposed to be commercialized, I do expect to be paid (as should anyone involved in the project).

Do you have any plans on how to sort this out?

**Comment by Tormentor667** [comment]
Currently I do not get any feedback and response from the company so I am not even sure if the contract or offer is still set.

If it will happen, there is still a portion of income that will be provided for the translators, promised - even if that means to reduce my piece of the cake. I am not that interested in earning money with it so I am fine with sharing my gain :)

**Comment by getphobo** [comment]
Well, mind that German translations of this kind have a market value of around €0.09-0.12 per source word, with €0.02-0.03 per word reviewed. Other languages may and do differ, but Western European languages generally sell at similar prices.

Anyway, at 60,000 words in the spreadsheet's English column, this would amount to some €40k for all eight languages to be translated and reviewed, minus what has already been done (you need to count the words, though, not the cells; for example, the currently reported 69% translated German is in fact only some 30% of the actual word count).

**Comment by Tormentor667** [comment]
I am aware that the translation is a valuable and very difficult task actually, and I do not expect someone to do it for free - instead we are very thankful that people volunteer to do it.

To make things clear once again: The free release of Blade of Agony will happen in any case, that’s what we have worked for - a free game powered by classic gaming enthusiasts.

The companies offer was to replace any asset with copyright for a commercial release, then sell it as a standalone game and return us 50% of the total income that we share between the developers. I am willed to offer a appropriate percentage to the translators as well, this is more than fair, but I can’t tell how much it is as long as the gaming company isn’t selling it neither answering my emails.

Therefore it doesn’t matter now and won’t matter anytime soon - Blade of Agony will be a free game, released for free by people who work on it for free - that’s our philosophy :) The commercial offer is just the cherry on the cake, nice to have but not essential to us. If by chance we all can make a bit of money, nice. If not, no one suffers from it.

**Comment by Tormentor667** [comment]
Closed for now until I receive information on further procedure from the company. Our priority is currently the free release.
